### PR TITLE
cmd/snap: make "snap find" error nicer.

### PR DESCRIPTION
**Pull request by chipaca**
also, cmd: delint; i18ize

This started as a trivial branch of me making the empty `snap find` error message nicer. And then I noticed some error messages weren't i18n'ed... and then I made lint complain less about things under `cmd`.

### CODE REVIEW

File: cmd/snap/cmd_changes.go [hunk 1]
```diff
@@ -70,11 +70,12 @@ func (c *cmdChanges) Execute(args []string) error {
 	}
 
 	if allDigits(c.Positional.Snap) {
-		return fmt.Errorf(`%s changes command expects a snap name, try: %[1]s change %s`, os.Args[0], c.Positional.Snap)
+		// TRANSLATORS: the first %s, and %[1]s, is the argv[0] (typically "snap"); the last %s is the argument given by the user to 'changes'
+		return fmt.Errorf(i18n.G(`%s changes command expects a snap name, try: %[1]s change %s`), os.Args[0], c.Positional.Snap)
```
Comment (chipaca) [inline cmd/snap/cmd_changes.go]: do we actually want this (in which case several other places need it), or should we just use "snap"?

File: cmd/snap/cmd_buy.go [hunk 2]
```diff
@@ -100,7 +100,7 @@ func buySnap(opts *store.BuyOptions) error {
 	}
 
 	// TODO Remove this payment method filter once interactive payment methods are supported on the CLI
-	methods := make([]*store.PaymentMethod, 0)
+	var methods []*store.PaymentMethod
```
Comment (zyga) [inline cmd/snap/cmd_buy.go]: Does it matter in any way if methods are `nil` in the code below?

File: cmd/snap/cmd_changes.go [hunk 3]
```diff
@@ -70,11 +70,12 @@ func (c *cmdChanges) Execute(args []string) error {
 	}
 
 	if allDigits(c.Positional.Snap) {
-		return fmt.Errorf(`%s changes command expects a snap name, try: %[1]s change %s`, os.Args[0], c.Positional.Snap)
+		// TRANSLATORS: the first %s, and %[1]s, is the argv[0] (typically "snap"); the last %s is the argument given by the user to 'changes'
```
Comment (zyga) [inline cmd/snap/cmd_changes.go]: <3

File: cmd/snap/cmd_find.go [hunk 4]
```diff
@@ -36,7 +37,7 @@ The find command queries the store for available packages.
 func getPrice(prices map[string]float64, currency string) (float64, string, error) {
 	// If there are no prices, then the snap is free
 	if len(prices) == 0 {
-		return 0, "", fmt.Errorf(i18n.G("snap is free"))
+		return 0, "", errors.New(i18n.G("snap is free"))
```
Comment (zyga) [inline cmd/snap/cmd_find.go]: I'd add # TRANSLATORS: Free as in gratis

File: cmd/snap/cmd_find.go [hunk 5]
```diff
@@ -100,6 +101,10 @@ func (x *cmdFind) Execute(args []string) error {
 		return ErrExtraArgs
 	}
 
+	if x.Positional.Query == "" {
+		return errors.New(i18n.G("you need to specify a query. For example: snap find hello-world"))
```
Comment (zyga) [inline cmd/snap/cmd_find.go]: (nitpick) `"you need to specify a query, e.g. \"snap find hello-world\""`

File: cmd/snap/main.go [hunk 6]
```diff
@@ -232,11 +232,10 @@ func run() error {
 		if e, ok := err.(*client.Error); ok && e.Kind == client.ErrorKindLoginRequired {
 			u, _ := user.Current()
 			if u != nil && u.Username == "root" {
-				return fmt.Errorf(`%s (see "snap login --help")`, e.Message)
```
Comment (zyga) [inline cmd/snap/main.go]: Is this removal intentional?

**Comment by zyga** [comment]
A few questions, thank you for doing this!

File: cmd/snap/cmd_find.go (hunk 4 above)
Comment (chipaca) [inline cmd/snap/cmd_find.go]: indeed!

File: cmd/snap/main.go (hunk 6 above)
Comment (chipaca) [inline cmd/snap/main.go]: i see no removal :-) but the diff is harder to read than it could. There are two things: the additon of `i18n.G` to both `fmt.Errorrf`s, and the outdent of the second `fmt.Errorf` prompted by golint

File: cmd/snap/cmd_buy.go (hunk 2 above)
Comment (chipaca) [inline cmd/snap/cmd_buy.go]: it's grown via `append`, and exits if its length is zero. No, it doesn't matter.

And: this change was suggested by golint.

File: cmd/snap/cmd_find.go (hunk 5 above)
Comment (chipaca) [inline cmd/snap/cmd_find.go]: i shall take this opportunity and make a unifying pass at all the little suggestions

File: cmd/snap/cmd_abort.go [hunk 7]
```diff
@@ -27,7 +27,7 @@ import (
 
 type cmdAbort struct {
 	Positional struct {
-		Id string `positional-arg-name:"change-id"`
+		ID string `positional-arg-name:"change-id"`
```
Comment (kyrofa) [inline cmd/snap/cmd_abort.go]: Ah, making the linter happy I see.

File: cmd/snap/cmd_booted.go [hunk 8]
```diff
@@ -48,13 +49,13 @@ func (x *cmdBooted) Execute(args []string) error {
...
 	if release.OnClassic {
-		fmt.Fprintf(Stdout, "Ignoring 'booted' on classic")
+		fmt.Fprintf(Stdout, i18n.G("Ignoring 'booted' on classic"))
 		return nil
 	}
 
 	bootloader, err := partition.FindBootloader()
 	if err != nil {
-		return fmt.Errorf("can not mark boot successful: %s", err)
+		return fmt.Errorf(i18n.G("can not mark boot successful: %s"), err)
```
Comment (kyrofa) [inline cmd/snap/cmd_booted.go]: We use "cannot" elsewhere.

File: cmd/snap/cmd_changes.go (hunk 1 above)
Comment (kyrofa) [inline cmd/snap/cmd_changes.go]: I like it. We've changed its name once before.

File: cmd/snap/cmd_find.go [hunk 9]
```diff
@@ -100,6 +102,10 @@ func (x *cmdFind) Execute(args []string) error {
 		return ErrExtraArgs
 	}
 
+	if x.Positional.Query == "" {
+		return errors.New(i18n.G("you need to specify a query. Try \"snap find hello-world\"."))
```
Comment (kyrofa) [inline cmd/snap/cmd_find.go]: Though if we're using `snap` elsewhere, like you mentioned, it may not be worth the ugliness.

**Comment by kyrofa** [comment]
You place handy `TRANSLATORS:` comments in some places with format strings, but not others. Should you do it for all? Or do you feel that some are simply clear enough?

**Comment by chipaca** [comment]
Yes, I try to only put them where there is lack of clarity on the meaning of the string in isolation of anything other than the project for which you're translating.

File: cmd/snap/cmd_find.go (hunk 9 above)
Comment (chipaca) [inline cmd/snap/cmd_find.go]: Yeah, we use it elsewhere and often in places where it would be very awkward to have something else (most likely: an absolute path to the binary instead of its bare name). And we haven't actually changed the name of a binary before; we added a new one, and removed the old one. Here's hoping we do it that way again if we do do it again.

File: i18n/i18n_test.go [hunk 10]
```diff
@@ -49,7 +49,7 @@ func (s *i18nTestSuite) SetUpTest(c *C) {
 	gettext.BindTextdomain("snappy-test", localeDir)
 	locale := gettext.SetLocale(gettext.LC_ALL, "en_DK.UTF-8")
 	if locale != "en_DK.UTF-8" {
-		c.Skip("can not init locale")
+		c.Skip("cannot init locale")
```
Comment (mvo5) [inline i18n/i18n_test.go]: Ha! Thanks for those!

**Comment by mvo5** [comment]
Looks very nice! :+1:
